### PR TITLE
E5 Flashing

### PR DESCRIPTION
**Pull request by jlpoltrack**
@olliw42 

Could you provide some feedback on this initial section for E5?  Thanks

To address: https://github.com/olliw42/mLRS-docu/issues/24

### CODE REVIEW

**Comment by olliw42** [comment]
MANY thx sir

I am not sure I had a detailed recipe in mind, I'm not sure I even know what I had in mind at all LOL. From the discussion linked to in the issue it deemed me that one could mention what general flashing procedures work or do not work. Like SWD: yes/no, USB-UART: yes/no, USB-DFU: yes/know, ???: yes/no, with maybe brief description of what to do or a link

not saying that more detailed recipoes wouldn't be highly useful, but I wonder if they should be repeated for each board, or if one shouldn't out-source them more into extra pages

the issue however indeed is that we have quite a number of different flash approaches for the different boards and modules, this is admittedly very confusing, and difficult to figure out for new users, like some have a boot button, some can be put into uart-systembootloader, some only provide SWD flashing, some can do uart-systembootloader but need a special tool, etc. pp.

**Review by 4O6FPV — COMMENTED** [review]
I think this is a great addition to the docs @olliw42 any way this can be approved so I can add onto this. I see it gives me the option to approve this not sure that's something you want the public able to do?

**Comment by olliw42** [comment]
@4O6FPV 
thx for considering contributing to the docs, this is much appreciated!!
@jlpoltrack is in fact the chief of ceremony for the mLRS-docu repo, we just agreed that we try to reach consensus but the final call is his. So, just to let you know that this question goes also to him. :)

>  I see it gives me the option to approve this not sure that's something you want the public able to do

this is probably indeed not what we want, And I'm surprised . Any idea which setting is wrong? 
(I'm really largely illiterate with regards to managing github repos)

do you indicate that you also have commit powers?

> any way this can be approved

I myself don't have any concerns. I think that some parts of the flashing are quite general and apply accros the various boards, and thus could go into a general page which can be references from the other boards too. But it is for sure an excellent start.

I guess the one change I would do is to move the flashing stuff to the end of the page. At least I would be more interested to first see what the boards would offer. But that's just my taste.

so, no block from me. LGTM.

File: docs/SEEEDSTUDIO_WIO_E5.md [hunk 1]
```diff
@@ -4,6 +4,33 @@
 
 The SeeedStudio [Wio-E5 module](https://wiki.seeedstudio.com/LoRa-E5_STM32WLE5JC_Module) is a highly attractive module for building mLRS equipment. SeeedStudio provides a number of boards which are based on this module, and which are quite interesting hardware for mLRS. However, these boards are not perfect since their pins are not ready-made for the purposes of mLRS. So, some tweaking and (easy) soldering is required.
 
+## Flashing the Modules ##
```
Comment (olliw42) [inline docs/SEEEDSTUDIO_WIO_E5.md:7]: I guess I would move that chapter to the end of the page. Just a suggestion.

File: docs/SEEEDSTUDIO_WIO_E5.md [hunk 2]
```diff
@@ -4,6 +4,33 @@
...
+- From the menu on the left select the Download tile
+- Select the correct firmware in the Download section, click Start Program
+- Power cycle the board, the red LED should blink which indicates that the board is disconnected
+
+### Subsequent Flashing ###
+
+Subsequent flashing can be done in two ways:
+1. Follow the initial flashing process
+    - There is no need to change the read out protection after the initial flash
+2. Boot into the system bootloader using the cli (or Lua)
```
Comment (olliw42) [inline docs/SEEEDSTUDIO_WIO_E5.md:32]: maybe "using the CLI (or the Lua script or the OLED display, if available)"

File: docs/SEEEDSTUDIO_WIO_E5.md [hunk 3]
```diff
@@ -4,6 +4,33 @@
...
+    - This enables programming over UART1 (PA9, PA10) or UART2 (PA2, PA3) with a USB<>UART adapter using the Serial connection option in STM32CubeProgrammer
```
Comment (olliw42) [inline docs/SEEEDSTUDIO_WIO_E5.md:33]: I typcially denote these devices as "USB-TTL adapters". There are some usb-serial adapters which are in fact inverting. Just a suggestion.

File: docs/SEEEDSTUDIO_WIO_E5.md [hunk 4]
```diff
@@ -4,6 +4,33 @@
 
 The SeeedStudio [Wio-E5 module](https://wiki.seeedstudio.com/LoRa-E5_STM32WLE5JC_Module) is a highly attractive module for building mLRS equipment. SeeedStudio provides a number of boards which are based on this module, and which are quite interesting hardware for mLRS. However, these boards are not perfect since their pins are not ready-made for the purposes of mLRS. So, some tweaking and (easy) soldering is required.
 
+## Flashing the Modules ##
+
+Important - every time that you power a board you should ensure that there is an antenna connected otherwise you risk damaging the RF section.
```
Comment (olliw42) [inline docs/SEEEDSTUDIO_WIO_E5.md:9]: maybe "Important: Every..:", 
maybe this shoul be highlighted, like bold and italics
just suggestions

**Comment by olliw42** [comment]
I just read through the original question and my first post in the above, and I think this indeed could be a long term plan of how to document the flashing. Just a suggestion, but at least a suggestion :)

We currently do have a couple methods (@brad112358 and I are think about ways to make this more canonical accros the boards):
- STLink
- system bootloader, via a BOOT button on power up
- system bootloader, via the BIND button on power up
- system bootloader, via the CLI/Lua/OLED

so, for each device there could be a table with these options, with three more colums "usable for initial flashing", "usuable for subsequent flashing" and "link to docu"

**Comment by olliw42** [comment]
> I see it gives me the option to approve this not sure that's something you want the public able to do

I have checked a box in the settings which I hope is doing what I hope it will do. 
@4O6FPV could you check/confirm that you are not able to commit, squash, or merge?
@jlpoltrack could you check/confirm that you have all the rights you need?

**Comment by 4O6FPV** [comment]
> > I see it gives me the option to approve this not sure that's something you want the public able to do
> 
> I have checked a box in the settings which I hope is doing what I hope it will do. @4O6FPV could you check/confirm that you are not able to commit, squash, or merge? @jlpoltrack could you check/confirm that you have all the rights you need? sorry for bothering you with this

looks like now I am not able to commit these changes or request changes. I can comment but the other two options are grayed out now.

File: docs/SEEEDSTUDIO_WIO_E5.md (hunk 1 above)
Comment (jlpoltrack) [reply docs/SEEEDSTUDIO_WIO_E5.md:7]: Done

File: docs/SEEEDSTUDIO_WIO_E5.md (hunk 2 above)
Comment (jlpoltrack) [reply docs/SEEEDSTUDIO_WIO_E5.md:32]: Added all three options

File: docs/SEEEDSTUDIO_WIO_E5.md (hunk 3 above)
Comment (jlpoltrack) [reply docs/SEEEDSTUDIO_WIO_E5.md:33]: Updated

File: docs/SEEEDSTUDIO_WIO_E5.md (hunk 4 above)
Comment (jlpoltrack) [reply docs/SEEEDSTUDIO_WIO_E5.md:9]: Now bold

**Comment by jlpoltrack** [comment]
@olliw42 
- I am still able to merge, so all good there.

@4O6FPV 
- Thanks for using these docs, hopefully they helped you along the way to use mLRS.

@olliw42  @4O6FPV 
- Let me know if there's any further feedback otherwise I will go ahead and merge today.

**Comment by olliw42** [comment]
> looks like now I am not able to commit these changes or request changes.

many thx for the feedback, so this click did not do the trick ... it is really sad that guithub can't provide better instructions ...

the checkbox says "_When enabled, only users explicitly granted access to this repository will be able to submit pull request reviews that "approve" or "request changes". All users able to submit comment pull request reviews will continue to be able to do so_." and in the collabroators section it says "_Direct access 1 has access to this repository_". But this one with access doens't actually have access ... what a nonsense

anyway, reverted it, should work now again as before

**Comment by olliw42** [comment]
MANY THX

LGTM :):):)

**Pull request merged**